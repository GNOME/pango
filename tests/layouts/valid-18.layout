{
  "context" : {
    "font" : "serif 12",
    "language" : "en-us",
    "base-gravity" : "south",
    "gravity-hint" : "natural",
    "base-dir" : "weak-ltr",
    "round-glyph-positions" : true,
    "transform" : [
      1,
      0,
      0,
      1,
      0,
      0
    ]
  },
  "comment" : "Miscellaneous ignorables",
  "text" : "some line breaks and mis­ce­llaneous ig﻿nora‌bles",
  "attributes" : [
    {
      "end" : 57,
      "type" : "show",
      "value" : 7
    }
  ],
  "font" : "Cantarell 11",
  "output" : {
    "is-wrapped" : true,
    "is-ellipsized" : false,
    "unknown-glyphs" : 5,
    "width" : 302080,
    "height" : 38912,
    "log-attrs" : [
      {
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "sentence-boundary" : true,
        "sentence-start" : true,
        "backspace-deletes-character" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "sentence-end" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "mandatory-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "sentence-boundary" : true,
        "sentence-start" : true,
        "backspace-deletes-character" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "expandable-space" : true,
        "word-boundary" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "backspace-deletes-character" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "backspace-deletes-character" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "line-break" : true,
        "mandatory-break" : true,
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "sentence-boundary" : true,
        "sentence-end" : true,
        "word-boundary" : true
      }
    ],
    "lines" : [
      {
        "start-index" : 0,
        "length" : 19,
        "paragraph-start" : true,
        "direction" : "ltr",
        "runs" : [
          {
            "offset" : 0,
            "length" : 16,
            "text" : "some line breaks",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "extra-attributes" : [
              {
                "end" : 57,
                "type" : "show",
                "value" : 7
              }
            ],
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 417,
                "width" : 7168,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 370,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 358,
                "width" : 13312,
                "is-cluster-start" : true,
                "log-cluster" : 2
              },
              {
                "glyph" : 287,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 3
              },
              {
                "glyph" : 1195,
                "width" : 11264,
                "is-cluster-start" : true,
                "log-cluster" : 4
              },
              {
                "glyph" : 349,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 5
              },
              {
                "glyph" : 324,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 6
              },
              {
                "glyph" : 360,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 7
              },
              {
                "glyph" : 287,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 8
              },
              {
                "glyph" : 1195,
                "width" : 11264,
                "is-cluster-start" : true,
                "log-cluster" : 9
              },
              {
                "glyph" : 272,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 10
              },
              {
                "glyph" : 409,
                "width" : 6144,
                "is-cluster-start" : true,
                "log-cluster" : 11
              },
              {
                "glyph" : 287,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 12
              },
              {
                "glyph" : 244,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 13
              },
              {
                "glyph" : 345,
                "width" : 7168,
                "is-cluster-start" : true,
                "log-cluster" : 14
              },
              {
                "glyph" : 417,
                "width" : 7168,
                "is-cluster-start" : true,
                "log-cluster" : 15
              }
            ]
          },
          {
            "offset" : 16,
            "length" : 3,
            "text" : " ",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "extra-attributes" : [
              {
                "end" : 57,
                "type" : "show",
                "value" : 7
              }
            ],
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 268443688,
                "width" : 17408,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          }
        ]
      },
      {
        "start-index" : 19,
        "length" : 38,
        "paragraph-start" : false,
        "direction" : "ltr",
        "runs" : [
          {
            "offset" : 19,
            "length" : 38,
            "text" : "and mis­ce­llaneous ig﻿nora‌bles",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "extra-attributes" : [
              {
                "end" : 57,
                "type" : "show",
                "value" : 7
              }
            ],
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 244,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 360,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 280,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 2
              },
              {
                "glyph" : 1195,
                "width" : 11264,
                "is-cluster-start" : true,
                "log-cluster" : 3
              },
              {
                "glyph" : 358,
                "width" : 13312,
                "is-cluster-start" : true,
                "log-cluster" : 4
              },
              {
                "glyph" : 324,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 5
              },
              {
                "glyph" : 417,
                "width" : 7168,
                "is-cluster-start" : true,
                "log-cluster" : 6
              },
              {
                "glyph" : 268435629,
                "width" : 23552,
                "is-cluster-start" : true,
                "log-cluster" : 7
              },
              {
                "glyph" : 273,
                "width" : 7168,
                "is-cluster-start" : true,
                "log-cluster" : 9
              },
              {
                "glyph" : 287,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 10
              },
              {
                "glyph" : 268435629,
                "width" : 23552,
                "is-cluster-start" : true,
                "log-cluster" : 11
              },
              {
                "glyph" : 349,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 13
              },
              {
                "glyph" : 349,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 14
              },
              {
                "glyph" : 244,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 15
              },
              {
                "glyph" : 360,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 16
              },
              {
                "glyph" : 287,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 17
              },
              {
                "glyph" : 370,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 18
              },
              {
                "glyph" : 438,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 19
              },
              {
                "glyph" : 417,
                "width" : 7168,
                "is-cluster-start" : true,
                "log-cluster" : 20
              },
              {
                "glyph" : 1195,
                "width" : 11264,
                "is-cluster-start" : true,
                "log-cluster" : 21
              },
              {
                "glyph" : 324,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 22
              },
              {
                "glyph" : 312,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 23
              },
              {
                "glyph" : 268500735,
                "width" : 23552,
                "is-cluster-start" : true,
                "log-cluster" : 24
              },
              {
                "glyph" : 360,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 27
              },
              {
                "glyph" : 370,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 28
              },
              {
                "glyph" : 409,
                "width" : 6144,
                "is-cluster-start" : true,
                "log-cluster" : 29
              },
              {
                "glyph" : 244,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 30
              },
              {
                "glyph" : 268443660,
                "width" : 17408,
                "is-cluster-start" : true,
                "log-cluster" : 31
              },
              {
                "glyph" : 272,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 34
              },
              {
                "glyph" : 349,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 35
              },
              {
                "glyph" : 287,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 36
              },
              {
                "glyph" : 417,
                "width" : 7168,
                "is-cluster-start" : true,
                "log-cluster" : 37
              }
            ]
          }
        ]
      }
    ]
  }
}
