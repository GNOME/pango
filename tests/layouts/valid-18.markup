
<span show="spaces|line-breaks|ignorables">some line breaks and mis­ce­llaneous ig﻿nora‌bles</span>
