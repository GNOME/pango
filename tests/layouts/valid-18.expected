some line breaks and mis­ce­llaneous ig﻿nora‌bles

--- parameters

wrapped: 1
ellipsized: 0
lines: 3

--- attributes

range 0 57
[0,57]show=7
range 57 2147483647

--- directions

0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 

--- cursor positions

0(0) 1(0) 2(0) 3(0) 4(0) 5(0) 6(0) 7(0) 8(0) 9(0) 10(0) 11(0) 12(0) 13(0) 14(0) 15(0) 16(0) 16(1) 20(0) 21(0) 22(0) 23(0) 24(0) 25(0) 26(0) 28(0) 29(0) 30(0) 32(0) 33(0) 34(0) 35(0) 36(0) 37(0) 38(0) 39(0) 40(0) 41(0) 42(0) 43(0) 46(0) 47(0) 48(0) 49(0) 53(0) 54(0) 55(0) 56(0) 56(1) 58(0) 

--- lines

i=1, index=0, paragraph-start=1, dir=ltr 'some line breaks '
i=2, index=19, paragraph-start=0, dir=ltr 'and mis­ce­llaneous ig﻿nora‌bles
'
i=3, index=58, paragraph-start=1, dir=ltr ''

--- runs

i=1, index=0, chars=16, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'some line breaks'
[0,57]show=7
i=2, index=16, chars=1, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, ' '
[0,57]show=7
i=3, index=19, no run, line end
i=4, index=19, chars=32, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'and mis­ce­llaneous ig﻿nora‌bles'
[0,57]show=7
i=5, index=57, no run, line end
i=6, index=58, no run, line end
